### ISSUE TITLE
[Feat]: [comm] Reliable delivery with ACK/NACK

### ISSUE THREAD

**Issue by purvanshjoshi** [sr-857/AstraGuard-AI#403]
### Is your feature request related to a problem?

Yes

HealthSummary and IntentMessages drop 10-20% during ISL congestion, breaking swarm coordination:

No ACK/NACK confirmation - silent packet loss undetected

No sequence numbers - cannot detect missing messages

No adaptive retry - fixed 1s intervals fail under jitter

10% loss acceptable? NO - leader election needs 100% critical messages

Current stats:

text
ISL 50-200ms latency + 10% loss = 1/10 HealthSummary dropped
→ Stale peer health → False leader elections → Swarm divergence

### Describe the solution you'd like

Add reliable delivery layer with sequence numbers + adaptive retry.

Features:

Sequence numbers per topic (health_seq: 12345)

ACK/NACK protocol - sender tracks pending messages

Adaptive retry: 1s → 2s → 4s → 8s (max 3 retries)

Delivery log for post-mortem analysis

API:

python
await bus.publish_reliable("coord", proposal, seq=12345)
# Auto-handles ACK/NACK + retry
Metrics:

text
End-to-end delivery: 99.9% 
Retry latency: <8s max
No duplicates under partition healing

### Domain

Backend (Node/FastAPI)
